docs/contribution-guides/Development-Standards-Guide.md

```diff
@@ -226,3 +242,301 @@ to just leave it as a public field without a property. As always, make sure to d
+```
+
+
+## 1. Writing modular code with minimal dependencies
+
+* Video games are event driven, you should use the `UpdateManager` as sparingly as possible and when it actually makes sense to constantly update a state of an object every frame.
+* Make use of `Action<>` and `UnityEvent<>` whenever you want to tell an object to do something based on a change or function being called on something else.
+
+_For more info about UnityEvents check: [The Offical Unity Docs](https://docs.unity3d.com/ScriptReference/Events.UnityEvent.html)_
+
+Events can be used to reduce the likelyhood of bugs and [NREs](https://learn.microsoft.com/en-us/dotnet/api/system.nullreferenceexception?view=net-7.0) occuring and makes our code much more maintainable and easy to expand upon due to how they're designed.
+
+With events, components don't have to constantly reference each other and only require a subscription to relevent functions that get invoked when things happen in the game.
+
+```cs
+//example of bad code
+
+class Health : Monobehavior
+{
+    public bool StateChanged;
+    public int currentHealth = 5;
+
+    void stateChange(int damageTaken){
+        currentHealth -= damageTaken;
+        if(currentHealth >= 0) return;
+        StateChanged = true;
+        Inventory.DropAll(); //How do we know that we have an inventory at all?
+    }
+}
+
+class DeathUI : Monobehavior
+{
+    public Health health; //<- possibily null
+
+    void Awake(){
+        UpdateManager.Add(FIXED_UPDATE, myFunc);
+    }
+
+    void myFunc(){
+        if(health.StateChanged == false) return;
+        DoThing();
+    }
+}
+
+class PlayerInventory : Inventory{
+    public void DropAll();
+}
+```
+
+```cs
+//example of good code
+
+class Health : Monobehavior
+{
+    public int CurrentHealth {private set; get;} = 5;
+    public Action OnDeath = new Action();
+
+    public void TakeDamage(int damageTaken)
+    {
+        currentHealth -= damageTaken;
+        if(currentHealth >= 0) return;
+        OnDeath?.Invoke();
+    }
+}
+
+class DeathUI : Monobehavior
+{
+    void OnEnable()
+    {
+        if(gameObject.TryGetComponent<Health>(out var playerHealth) == false) 
+        {
+            // Always leave warnings and errors whenever possible instead of silently not reporting unintended behaviors.
+            Logger.LogWarning("[DeathUI/OnEnable] - No health found on this gameObject! DeathUI hasn't subscribed to any events when it got enabled!");
+            return;
+        }
+        // Subscribes to the OnDeath event which will trigger when this object's health reaches 0 or below.
+        playerHealth.OnDeath += myFunc;
+    }
+
+    void myFunc()
+    {
+        DoThing();
+    }
+}
+
+class PlayerInventory : Inventory
+{
+    //same thing as in DeathUI, just subscribe to the OnDeath event.
+}
+```
+
+_Note: Events can leak! Always remember to unsubscribe your events on `OnDisable()`._
+
+* Minimise Polymorphism, Embrace Interfaces.
```

Comment:
You mean inheritance, not polymorphism.

Reply:
True. I only thought about polymorphism during that instance so my brain defaulted to that. Changing.